eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -29,35 +29,92 @@ jobs:
           OSVmImage: 'macOS-10.15'
           vcpkg.deps: 'curl[ssl]'
           VCPKG_DEFAULT_TRIPLET: 'x64-osx'
-          build.args: ' -DTRANSPORT_CURL=ON  -DAZ_PLATFORM_IMPL=POSIX'
-
+          build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX'
+        Win_x64_with_iot_samples:
+          OSVmImage: 'windows-2019'
+          vcpkg.deps: 'curl[winssl] cmocka paho-mqtt'
+          VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
+          CMAKE_GENERATOR: 'Visual Studio 16 2019'
+          CMAKE_GENERATOR_PLATFORM: x64
+          build.args: ' -DTRANSPORT_PAHO=ON'
+          test_type: 'iot'
+        Linux_x64_with_iot_samples:
+          OSVmImage: 'ubuntu-18.04'
+          vcpkg.deps: 'curl[ssl] cmocka paho-mqtt'
+          VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+          build.args: ' -DTRANSPORT_PAHO=ON'
+          test_type: 'iot'
     pool:
       vmImage: $(OSVmImage)
 
     steps:
       - template: /eng/pipelines/templates/steps/vcpkg.yml
         parameters:
           DependenciesVariableName: vcpkg.deps
+          
+      - template: /eng/common/TestResources/deploy-test-resources.yml
+        parameters:
+          ServiceDirectory: '.'
+
+      - pwsh: |
```

Comment:
You should consider creating a script for this so it is a little easier to debug and run locally. Usually large inline script blocks because very difficult to debug if they fail.